src/sigmatcher/analysis.py

```diff
@@ -219,28 +178,75 @@ def analyze(self, unpacked_path: Path, results: Dict["Analyzer", Union[Result, E
         parent_class_result.matched_methods.append(matched_method)
         return matched_method
 
+    def get_dependencies(self) -> Set[str]:
+        return self.definition.get_dependencies() | {self.parent.name}
+
     @property
     def name(self) -> str:
         return f"{self.parent.name}.methods.{self.definition.name}"
 
 
-def analyze(definitions: Definitions, unpacked_path: Path) -> None:
-    analyzers: List[Analyzer] = []
+@dataclasses.dataclass(frozen=True)
+class ExportAnalyzer(Analyzer):
+    definition: ExportDefinition
+    parent: ClassAnalyzer
+
+    def analyze(self, unpacked_path: Path, results: Dict[str, Union[Result, Exception, None]]) -> MatchedExport:
+        parent_class_result = results[self.parent.name]
+        assert isinstance(parent_class_result, MatchedClass)
+
+        raw_class = parent_class_result.smali_file.read_text()
+        signature = self.definition.signatures[0].resolve_macros(results)
+        captured_names = set(signature.capture(raw_class))
+        check_match_count(captured_names, self.name)
+        export_value = next(iter(captured_names))
+
+        return MatchedExport.from_value(export_value)
+
+    def get_dependencies(self) -> Set[str]:
+        return self.definition.get_dependencies() | {self.parent.name}
+
+    @property
+    def name(self) -> str:
+        return f"{self.parent.name}.exports.{self.definition.name}"
+
+
+def create_analyzers(definitions: Definitions) -> Dict[str, Analyzer]:
+    name_to_analyzer: Dict[str, Analyzer] = {}
     for class_definition in definitions.defs:
-        class_analyzer = ClassAnalyzer(class_definition, dependencies=())
-        analyzers.append(class_analyzer)
+        class_analyzer = ClassAnalyzer(class_definition)
+        name_to_analyzer[class_definition.name] = class_analyzer
+
         for method_definition in class_definition.methods:
-            analyzers.append(MethodAnalyzer(method_definition, dependencies=(class_analyzer,), parent=class_analyzer))
+            method_analyzer = MethodAnalyzer(method_definition, parent=class_analyzer)
+            name_to_analyzer[method_analyzer.name] = method_analyzer
+
+        for field_definition in class_definition.fields:
+            field_analyzer = FieldAnalyzer(field_definition, parent=class_analyzer)
+            name_to_analyzer[field_analyzer.name] = field_analyzer
+
         for field_definition in class_definition.fields:
-            analyzers.append(FieldAnalyzer(field_definition, dependencies=(class_analyzer,), parent=class_analyzer))
+            field_analyzer = FieldAnalyzer(field_definition, parent=class_analyzer)
+            name_to_analyzer[field_analyzer.name] = field_analyzer
```

Comment:
Twice?

Reply:
guessing you meant exports and copy-pasted badly

Reply:
I sure did :blush: